### PR TITLE
Add tests for lwm2m_list

### PR DESCRIPTION
**Pull request by LukasWoodtli**
Added some basic unit tests for the linked list implementation.

### CODE REVIEW

**Review by rettichschnidi — CHANGES_REQUESTED** [review]

File: tests/list_tests.c
[the remark this answers is not in the document]
Comment (LukasWoodtli) [reply tests/list_tests.c:1]: Test files don't have a header in the Wakaama project. The test suite functions are just declared in `tests.h`.

File: tests/list_tests.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
+#include "CUnit/Basic.h"
+#include "internals.h"
+#include "liblwm2m.h"
+#include "tests.h"
+
+typedef struct {
+    struct list_node_ *next; // matches lwm2m_list_t::next
+    uint16_t mID;            // matches lwm2m_list_t::id
```
[the remark this answers is not in the document]
Comment (LukasWoodtli) [reply tests/list_tests.c:7]: Maybe we could replace the `next` and the `mID` variables in the struct with a single `lwm2m_list_t` variable. But in the rest of the code it's done like here. So it's more consistent with the rest of the codebase.

File: tests/list_tests.c
[the remark this answers is not in the document]
Comment (rettichschnidi) [reply tests/list_tests.c:1]: Arg - what I meant is they should have a copyright header. .)

File: tests/list_tests.c [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    for (uint_fast8_t i = 0; i < 5; ++i) {
+        list_node_t *node = lwm2m_malloc(sizeof(list_node_t));
+        node->mID = i;
+        LWM2M_LIST_ADD(list, node);
+    }
+
+    lwm2m_list_t *ret = LWM2M_LIST_FIND(list, 1);
+    CU_ASSERT_PTR_NOT_NULL(ret);
+
+    LWM2M_LIST_FREE(list);
```
Comment (rettichschnidi) [inline tests/list_tests.c:82]: With the assertion gone it is no longer obvious if and how this test actually finds out whether LWM2M_LIST_FREE() actually did something.

Maybe add a comment along the lines of "Functionality implicitly tested by leak sanitizer".

**Review by rettichschnidi — CHANGES_REQUESTED** [review]

File: tests/list_tests.c (hunk 2 above)
Comment (LukasWoodtli) [reply tests/list_tests.c:82]: Good point!

**Pull request merged**